app/src/main/java/com/ciechu/kursakademiaandroida/core/base/UseCase.kt

```diff
@@ -0,0 +1,20 @@
+package com.ciechu.kursakademiaandroida.core.base
+
+import kotlinx.coroutines.*
+
+abstract class UseCase<out Type, in Params> {
+
+    abstract suspend fun action(params: Params): Type
+
+    operator fun invoke(scope: CoroutineScope,
+                        params: Params,
+                        executionDispatcher: CoroutineDispatcher = Dispatchers.IO,
+                        onResults: (Result<Type>) -> Unit = {}) {
```

Comment:
Sugerowałbym tutaj nazwanie tego `onResult` bo nawet jeśli będziemy pobierać listę obiektów, to będzie ona opakowana zawsze w pojedynczy Result.